### ISSUE TITLE
Problem with using fexpr~

### ISSUE THREAD

**Issue by bertrandopiroscafo** [monkeyswarm/MobMuPlat#60]
Hello,
I use a Pure Data patch using the fexpr~ object for implementing the Karplus-Strong Algorithm.
I have the following errors on iOS:
fexpr~:expression:type mismatch
Verbose(4): ... you might be able to track this down from the Find Menu
error:expr: no method for 'tilts'
I have also errors for others system messages like '/gyro' and so on but I do not route any of these messages. After a while, the app crashes. 

I have an another question: what is the difference between 'route' and 'list trim'? 
Thank you for your help!
@bertrandopiroscafo on IG

**Comment by monkeyswarm** [comment]
Please attach the PD patch and I can take a look.

**Comment by bertrandopiroscafo** [comment]
[BUG_SUBMITTED.zip](https://github.com/monkeyswarm/MobMuPlat/files/2971798/BUG_SUBMITTED.zip)
Hello!
This is the entire patch. Thank you very much to take a look.
Thanks to MobMuPlat, I build iPad modules which I can integrate in a modular synth. Great!

**Comment by monkeyswarm** [comment]
I get this error on both MobMuPlat and also when I run the patch on desktop Pure Data vanilla .48-1, so it's not an issue specific to MobMuPlat. It ran on my iphone 6s / iOS 12 for several minutes without incident.

Please make sure you are using the most recent version of MobMuPlat, it's possible that there were updates to fexpr in a recent Pd version which were recently pulled into MobMuPlat. (Which version of Pd did you develop the patch on?)

======start console log
fexpr~: bad output index, (-0.000000)
$y1
-
$f2 


fexpr~: no error report till next reset
fexpr~: index assumed to be = -1
=======end console log

**Comment by bertrandopiroscafo** [comment]
Thank you for your reply. I get also this error when the patch is running on my Mac but it does not seem to be a fatal error (I have nevertheless to dig it).
I use the most recent version of MobMuPlat and Vanilla .48-1.

But do you observe the following message:

fexpr~:expression:type mismatch
Verbose(4): ... you might be able to track this down from the Find Menu
error:expr: no method for 'tilts'

It seems that this message is specific to MobMuPlat because there aren't system messages of this type on a computer. Do I think wrong?

**Comment by bertrandopiroscafo** [comment]
I run on iOS 9.3.5

**Comment by monkeyswarm** [comment]
I do not observe that. The patch you sent me does not have any
[receive fromSystem]
objects in it, which is where /tilt and /gyro and other sensor data flow.

**Comment by bertrandopiroscafo** [comment]
That's right. I will make a try on a more recent iOS. I let you know if I observe a different behavior. Thanks a lot for your support!

**Comment by monkeyswarm** [comment]
I'm saying I think you sent me a different version of the Pd patch than the one that is showing your tilt/gyro messages on the console.

**Comment by bertrandopiroscafo** [comment]
Hello Daniel,
I made others tests during the weekend. With the simplest patch MMPTutorial0-HelloSine.pd, I observe the error messages mentioned above when I tweak the pitch while I move and shake my iPad mini1 (iOS 9.3.5) and my iPad mini2 (iOS 11.4.1). It can take a while before I can see the error messages. Do you observe the same phenomenon ? I use the latest version of MobMuPlat app and the latest version of the editor.

Thank you for your help!
Bertrand
